src/thumbnailer.cc

```diff
@@ -39,16 +39,81 @@ Thumbnailer::~Thumbnailer() { WebPAnimEncoderDelete(enc_); }
 Thumbnailer::Status Thumbnailer::AddFrame(const WebPPicture& pic,
                                           int timestamp_ms) {
   // Verify dimension of frames.
-  if (!frames_.empty() && (pic.width != (frames_[0].pic).width ||
-                           pic.height != (frames_[0].pic).height)) {
+  if (!frames_.empty() && (pic.width != frames_[0].pic.width ||
+                           pic.height != frames_[0].pic.height)) {
     return kImageFormatError;
   }
   WebPConfig new_config;
   WebPConfigInit(&new_config);
+  new_config.show_compressed = 1;
   frames_.push_back({pic, timestamp_ms, new_config});
   return kOk;
 }
 
+int Thumbnailer::GetSize(WebPPicture* const pic, const WebPConfig& config) {
+  const int failure = -1;
+
+  WebPPicture new_pic;
+  WebPPictureInit(&new_pic);
```

Comment:
same here: this call can (rarely) fail. Could be incorporated with the if(...) below.

Reply:
actually, you don't need this call to WebPPictureInit() if you're calling WebPPictureCopy() directly